src/script/conversation/ConversationRepository.ts

```diff
@@ -2978,10 +2965,10 @@ export class ConversationRepository {
    *
    * @param conversationEntity Conversation fetch events and users for
    */
-  private fetchUsersAndEvents(conversationEntity: Conversation) {
+  private async fetchUsersAndEvents(conversationEntity: Conversation) {
     if (!conversationEntity.is_loaded() && !conversationEntity.is_pending()) {
-      this.updateParticipatingUserEntities(conversationEntity);
-      this.getUnreadEvents(conversationEntity);
+      await this.updateParticipatingUserEntities(conversationEntity);
+      await this.getUnreadEvents(conversationEntity);
```

Comment:
Here we load the user entities and the events from the DB into the in-memory conversation. But we do not wait for the operation to be done before returning (thus the glitch)